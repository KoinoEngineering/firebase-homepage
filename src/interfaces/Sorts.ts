/** 比較ソート用インターフェース */
export interface CompoersionSort {
  /** 配列 */
  items: CompoersionSortItem[];
  /** 現在地 */
  cursor: number;
  /** カーソルの最大最小 */
  cursorMin: number;
  cursorMax: number;

  /** 表示上必要な比較対象を表すカーソル */
  comparison: number;

  /** シェーカーソートなど向きのあるもので使う向き */
  direction: boolean;

  /** minやmax等を入れる入れ物 */
  pointer: number;
  /**
   * 比較と入れ替えを別ステップでやりたいので
   * やむを得ず次のステップで入れ替えるかどうかを記憶させる
   */
  needSwap: boolean;

  /** 比較回数 */
  compCnt: number;
  /** 入れ替え回数 */
  swapCnt: number;

  /** 完了しているかどうか */
  ended: boolean;
}

/** 比較ソートの中身 */
export interface CompoersionSortItem {
  id: string;
  value: number;
}

/** ソートの種類 */
export enum SORT_TYPES {
  bubble = "バブルソート",
}
