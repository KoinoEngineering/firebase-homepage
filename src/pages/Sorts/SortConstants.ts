export enum ORDER {
    ASC = "asc",
    DESC = "desc"
}

export const ORDER_LABEL_MAP = {
    [ORDER.ASC]: "昇順",
    [ORDER.DESC]: "降順",
};
