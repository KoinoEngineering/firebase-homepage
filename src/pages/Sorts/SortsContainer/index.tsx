import { makeStyles, createStyles } from "@material-ui/core";
import React from "react";
import { ComparsionItem } from "src/models/sorts/Comparsion";
import defaultTheme from "src/utils/theme";

interface SortsContainerProps {
    items: ComparsionItem[];
    ended: boolean;
}

const useStyles = makeStyles<typeof defaultTheme, { value: number; fixed: boolean }>(
    createStyles({
        rainbow: ({ value, fixed }) => ({
            backgroundColor: `hsl(${(value * 360) / 100},${fixed ? "50%" : "80%"},${fixed ? "50%" : "80%"})`,
            display: "inline-block",
        }),
        sqare: {
            width: "2rem",
            "&::before": {
                content: "\"\"",
                paddingTop: "100%",
                display: "block",
            },
        },
        relative: {
            position: "relative",
        },
        absolute: {
            position: "absolute",
        },
        content: {
            top: 0,
            left: 0,
            paddingTop: "50%",
            paddingLeft: "50%",
            marginTop: "-25%",
            marginLeft: "-25%",
        },
    }),
);

const ContainerItem: React.FC<{ value: number; fixed: boolean }> = ({ value, fixed }) => {
    const { rainbow, sqare, relative, absolute, content } = useStyles({ value, fixed });
    return (
        <div className={[rainbow, sqare, relative].join(" ")}>
            <div className={[absolute, content].join(" ")}>{value}</div>
        </div>
    );
};

const SortsContainer: React.FC<SortsContainerProps> = ({ items, ended }) => {
    return (
        <div>
            {items.map(({ id, value, fixed }) => {
                return <ContainerItem key={id} value={value} fixed={!ended && fixed} />;
            })}
        </div>
    );
};

export default SortsContainer;
