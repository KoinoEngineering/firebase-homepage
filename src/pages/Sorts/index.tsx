import { Grid, Typography } from "@material-ui/core";
import React, { useMemo } from "react";
import { useDispatch } from "react-redux";
import HeadlineCard, { HeadlineCardProps } from "src/components/templates/HeadlineCard";
import PageContainer from "src/components/templates/Page/PageContainer";
import { navigateActionsCreatetors } from "src/utils/ComponentUtils";
import { bindActionCreators } from "redux";
import ROUTES from "src/utils/routes";

const Sorts: React.FC = () => {

    const dispatch = useDispatch();
    const actions = useMemo(() => ({
        navigate: bindActionCreators(navigateActionsCreatetors, dispatch)
    }), [dispatch]);
    const sorts: HeadlineCardProps[] = Object.entries(SORTS_TITLE_MAP).map(([route, { title, exist }]) => {
        return {
            onClick: exist ? () => actions.navigate.push(route) : undefined,
            cardActionAreaProps: {
                disableRipple: !exist,
                headerProps: {
                    title: title + (exist ? "" : "(Comming Soon...)")
                }
            }
        };
    });

    return <PageContainer id="Sorts">
        <Grid id="title" container>
            <Grid container>
                <Typography variant="h2">
                    いろんなソートを可視化してみる
                </Typography>
            </Grid>
            <Grid container>
                <Typography variant="h3">
                    制約としてループを使用しない
                </Typography>
            </Grid>
        </Grid>
        <Grid id="main" container spacing={4}>
            {
                sorts.map(sort => {
                    return <Grid key={JSON.stringify(sort)} item>
                        <HeadlineCard {...sort} />
                    </Grid>;
                })
            }
        </Grid>
    </PageContainer >;
};

export default Sorts;

type SortTitleMap = {
    [K: string]: {
        title: string;
        exist?: boolean;
    };
};

export const SORTS_TITLE_MAP: SortTitleMap = {
    [ROUTES.SORTS_BUBBLE]: { title: "バブルソート", exist: true },
    [ROUTES.SORTS_SHAKER]: { title: "シェーカーソート", exist: true },
    [ROUTES.SORTS_GNOME]: { title: "ノームソート" },
    [ROUTES.SORTS_SELECT]: { title: "選択ソート" },
    [ROUTES.SORTS_INSERT]: { title: "挿入ソート" },
    [ROUTES.SORTS_BACKET]: { title: "バケットソート" },
    [ROUTES.SORTS_MARGE]: { title: "マージソート" },
    [ROUTES.SORTS_HEAP]: { title: "ヒープソート" },
    [ROUTES.SORTS_QUICK]: { title: "クイックソート" },
};