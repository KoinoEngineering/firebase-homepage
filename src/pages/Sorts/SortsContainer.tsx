import { createStyles, makeStyles } from "@material-ui/core";
import React from "react";
import { ComparisonSort, ComparisonSortItem } from "src/interfaces/Sorts";
import defaultTheme from "src/utils/theme";

const useStyles = makeStyles<typeof defaultTheme, { value: number; fixed: boolean; max: number }>(
    createStyles({
        absolute: {
            position: "absolute",
        },
        compared: {
            borderColor: "red",
            borderStyle: "solid",
            borderWidth: 3,
        },
        content: {
            left: "50%",
            top: "50%",
            transform: "translateX(-50%) translateY(-50%)",
        },
        inlineBlock: {
            display: "inline-block",
        },
        pined: {
            borderColor: "blue",
            borderStyle: "solid",
            borderWidth: 3,
        },
        rainbow: ({ value, fixed, max }) => ({
            backgroundColor: `hsl(${(value * 360) / max},${fixed ? "50%" : "80%"},${fixed ? "50%" : "80%"})`,
        }),
        relative: {
            position: "relative",
        },
        sqare: {
            "&::before": {
                content: "\"\"",
                display: "block",
                paddingTop: "100%",
            },
            width: "2rem",
        },
    }),
);

interface ContainerItemProps {
    item: ComparisonSortItem;
    ended: boolean;
    isCursor: boolean;
    isPointer: boolean;
    max: number;
}

const ContainerItem: React.FC<ContainerItemProps> = ({ item: { value }, isCursor, ended, isPointer, max }) => {
    const { inlineBlock, compared, pined, rainbow, sqare, relative, absolute, content } = useStyles({ fixed: ended, max, value });
    return (
        <div className={[inlineBlock, rainbow, isPointer ? pined : isCursor ? compared : "", sqare, relative].join(" ")}>
            <div className={[absolute, content].join(" ")}>{value}</div>
        </div>
    );
};

const SortsContainer: React.FC<ComparisonSort> = ({ items, ended, cursor, pointer, comparison }) => {
    return (
        <div>
            {items.map((item, i) => {
                return (
                    <ContainerItem
                        key={item.id}
                        item={item}
                        ended={ended}
                        isCursor={i === cursor || i === comparison}
                        isPointer={i === pointer}
                        max={items.length}
                    />
                );
            })}
        </div>
    );
};

export default SortsContainer;
