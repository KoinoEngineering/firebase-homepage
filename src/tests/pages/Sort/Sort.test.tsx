import React from "react";
import { renderWithProvider } from "src/tests/tetutils";
import Sorts from "src/pages/Sorts";

test("Sorts expect to match snapshot", () => {
    const target = renderWithProvider(<Sorts />);
    expect(target).toMatchSnapshot();
});
