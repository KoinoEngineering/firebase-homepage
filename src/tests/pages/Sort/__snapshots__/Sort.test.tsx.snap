// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorts expect to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root SortsPageContainer-root-2 MuiContainer-maxWidthLg"
        id="Sorts"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
          id="title"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              いろんなソートを可視化してみる
            </h2>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3"
            >
              制約としてループを使用しない
            </h3>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
          id="main"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      バブルソート
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      シェーカーソート
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      ノームソート
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      選択ソート
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      挿入ソート(Comming Soon...)
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      バケットソート(Comming Soon...)
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      マージソート(Comming Soon...)
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      ヒープソート(Comming Soon...)
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      クイックソート(Comming Soon...)
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root SortsPageContainer-root-2 MuiContainer-maxWidthLg"
      id="Sorts"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
        id="title"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2"
          >
            いろんなソートを可視化してみる
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            制約としてループを使用しない
          </h3>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
        id="main"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    バブルソート
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    シェーカーソート
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    ノームソート
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    選択ソート
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    挿入ソート(Comming Soon...)
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    バケットソート(Comming Soon...)
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    マージソート(Comming Soon...)
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    ヒープソート(Comming Soon...)
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    クイックソート(Comming Soon...)
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
